From ba5dea982d2ad2bae18403bac68a8d6d516ae766 Mon Sep 17 00:00:00 2001
From: Davidlohr Bueso <dbueso@suse.de>
Date: Thu, 12 Mar 2015 08:45:52 -0700
Subject: [PATCH] locking: Update atomic rmw flags
Git-commit: 
Patch-mainline: None, not it mainline
References: 

The last refresh from 2709045073e cause some archs to
not have this flag, most noticibly x86, update accordingly.

Signed-off-by: Davidlohr Bueso <dbueso@suse.de>

---
 arch/s390/Kconfig |    1 +
 arch/tile/Kconfig |    1 +
 arch/x86/Kconfig  |    2 ++
 3 files changed, 4 insertions(+)

--- a/arch/s390/Kconfig
+++ b/arch/s390/Kconfig
@@ -117,6 +117,7 @@ config S390
 	select ARCH_INLINE_WRITE_UNLOCK_BH
 	select ARCH_INLINE_WRITE_UNLOCK_IRQ
 	select ARCH_INLINE_WRITE_UNLOCK_IRQRESTORE
+	select ARCH_SUPPORTS_ATOMIC_RMW
 
 config SCHED_OMIT_FRAME_POINTER
 	def_bool y
--- a/arch/tile/Kconfig
+++ b/arch/tile/Kconfig
@@ -13,6 +13,7 @@ config TILE
 	select GENERIC_IRQ_SHOW
 	select HAVE_SYSCALL_WRAPPERS if TILEGX
 	select SYS_HYPERVISOR
+	select ARCH_SUPPORTS_ATOMIC_RMW
 
 # FIXME: investigate whether we need/want these options.
 #	select HAVE_IOREMAP_PROT
--- a/arch/x86/Kconfig
+++ b/arch/x86/Kconfig
@@ -70,6 +70,8 @@ config X86
 	select IRQ_FORCED_THREADING
 	select USE_GENERIC_SMP_HELPERS if SMP
 	select HAVE_BPF_JIT if (X86_64 && NET)
+	select ARCH_SUPPORTS_ATOMIC_RMW
+	select ARCH_HAVE_NMI_SAFE_CMPXCHG
 
 config INSTRUCTION_DECODER
 	def_bool (KPROBES || PERF_EVENTS)
